agent/api/task/task.go

```diff
@@ -1760,23 +1760,47 @@ func (task *Task) PopulateSecretsAsEnv(container *apicontainer.Container) *apier
 
 	envVars := make(map[string]string)
 
+	logDriverTokenName := ""
+	logDriverTokenSecretValue := ""
+
 	for _, secret := range container.Secrets {
-		if secret.Provider == apicontainer.SecretProviderSSM && secret.Type == apicontainer.SecretTypeEnv {
-			k := secret.GetSecretResourceCacheKey()
-			if secretValue, ok := ssmRes.GetCachedSecretValue(k); ok {
-				envVars[secret.Name] = secretValue
-			}
+		k := secret.GetSecretResourceCacheKey()
+
+		secretVal := ""
+		validValue := false
+
+		if secretValue, ok := ssmRes.GetCachedSecretValue(k); ok {
+			validValue = ok
+			secretVal = secretValue
+		}
+		if secretValue, ok := asmRes.GetCachedSecretValue(k); ok {
+			validValue = ok
+			secretVal = secretValue
 		}
 
-		if secret.Provider == apicontainer.SecretProviderASM && secret.Type == apicontainer.SecretTypeEnv {
-			k := secret.GetSecretResourceCacheKey()
-			if secretValue, ok := asmRes.GetCachedSecretValue(k); ok {
-				envVars[secret.Name] = secretValue
-			}
+		if !validValue {
+			continue
 		}
+
+		if secret.Type == apicontainer.SecretTypeEnv {
+			envVars[secret.Name] = secretVal
+			continue
+		}
+		if secret.Target == apicontainer.SecretTargetLogDriver {
+			logDriverTokenName = secret.Name
```

Comment:
i think currently you assume there's only one log driver secret, but i don't see why you need to have that assumption.. i think you can call ApplyLogDriverSecret here whenever you find a log driver secret?